tests/api/test_devicespec.py

```diff
@@ -213,6 +213,8 @@ def test_invalid_spec(self):
             {"phase_0": 1.23},
             {"phase_0": 0},
             {"phase_0": 6.29999},
+            {"phase_0": [6.29999, 0, 0.14]},
+            {"phase_0": [[0.1, 0, 2], [[0.5]]]},
```

Comment:
yep, recursion for the deeply nested ones ♻️🤓

In which cases the TDM program uses arrays instead of single parameter values?